src/control/positionControl/pathPlanning/NumTreesPlanning.cpp

```diff
@@ -34,12 +34,16 @@ NumTreesPlanning::computePath(const Vector2 &robotPosition, const Vector2 &targe
             }
             auto parentCollision = collisionDetector.getCollisionBetweenPoints(point.getParent()->getPosition(), point.getPosition());
             if (parentCollision){
-                auto branches = branchPath(*point.getParent(), parentCollision.value());
+                auto branches = branchPath(*point.getParent(), parentCollision.value(), targetPosition);
                 std::for_each(branches.begin(), branches.end(), [&pointQueue](PathPointNode& newPoint){
                     pointQueue.push(newPoint);
                     });
                 continue;
             }
+            // current path has no collisions - update current best path
+            if (point.getPosition() - targetPosition < finalPath.getPosition() - targetPosition){
```

Comment:
What? I did not know that comparisons are defined for Vector2?

Reply:
The < operator is defined. It compares the length of 2 vectors